### ISSUE TITLE
Support Recurrent Batch Normalization

### ISSUE THREAD

**Issue by beam2d** [chainer/chainer#1085]
See http://arxiv.org/abs/1603.09025. It can be built with existing building blocks, though the implementation may be a bit complicated (in particular for handling different BatchNormalization moving statistics for different time steps), and require much memory (since F.lstm cannot be used). So it is better to design new one.

**Comment by unnonouno** [comment]
:+1:

**Comment by MannyKayy** [comment]
Here is a reference chainer implementation of [recurrent batch normalization](https://github.com/musyoku/recurrent-batch-normalization)

**Comment by MannyKayy** [comment]
bump
